### PR TITLE
Bugfix for applications with MongoDB datasources authenticated with SRV string

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

When we try to fork or import an application containing the MongoDB datasource authenticated with SRV string, the destination application breaks with an authentication failure error. 

Fixes :
#4936
#4929

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually tested on local and release instances.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Existing tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java [hunk 1]
```diff
@@ -403,10 +422,14 @@ private boolean updateOnLoadActionsWithNewActionIds(Map<String, String> actionId
...
                                 makePristine(templateDatasource);
-
                                 templateDatasource.setOrganizationId(toOrganizationId);
-
-                                return createSuffixedDatasource(templateDatasource);
+                                DatasourceConfiguration datasourceConfig = templateDatasource.getDatasourceConfiguration();
+                                return pluginService.findById(templateDatasource.getPluginId())
+                                    .flatMap(plugin -> {
+                                        
+                                        updateSrvForMongoDatasource(plugin, datasourceConfig);
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:430]: wouldn't this function be invoked for any non mongo plugin as well ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java [hunk 2]
```diff
@@ -458,6 +481,35 @@ private boolean updateOnLoadActionsWithNewActionIds(Map<String, String> actionId
                     throw error;
                 });
     }
+    
+    /**
+     * @param plugin to check if the plugin-type is mongo-plugin
+     * @param datasourceConfig to update Property object
+     */
+    public void updateSrvForMongoDatasource(Plugin plugin, DatasourceConfiguration datasourceConfig) {
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:489]: this seems like a plugin specific operation. Can we move this to appsmith-plugin package somehow ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:430]: Yes as we don't have the plugin type upfront and as we wanted to check for MongoDB datasource specifically we need the respective plugin object. Any suggestions on this?

Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ExamplesOrganizationCloner.java:430]: The way I can see it right now is this - define a method say 'foo' in the common plugin interface. This method should do nothing for other plugins but for mongo plugin this method can be overridden to do any required transformations. @sharat87 can you see any other way to doing this ?

**Review by sumitsum — APPROVED** [review]

**Pull request merged**